src/ruby/lib/grpc/generic/rpc_server.rb

```diff
@@ -407,7 +413,13 @@ def loop_handle_server_calls
       request_call_tag = Object.new
       until stopped?
         deadline = from_relative_time(@poll_period)
-        an_rpc = @server.request_call(@cq, request_call_tag, deadline)
+        begin
+          an_rpc = @server.request_call(@cq, request_call_tag, deadline)
+        rescue Core::CallError, RuntimeError => e
+          # can happen during server shutdown
```

Comment:
The server will report success == 0 for any requested calls at shutdown
time.

The relevant core test is here:

https://github.com/grpc/grpc/blob/master/test/core/end2end/tests/early_server_shutdown_finishes_tags.c

On Fri, May 22, 2015, 5:08 PM Tim Emiola <notifications@github.com> wrote:

> In src/ruby/lib/grpc/generic/rpc_server.rb
> <https://github.com/grpc/grpc/pull/1735#discussion_r30939122>:
>
> > @@ -407,7 +413,13 @@ def loop_handle_server_calls
> >        request_call_tag = Object.new
> >        until stopped?
> >          deadline = from_relative_time(@poll_period)
> > -        an_rpc = @server.request_call(@cq, request_call_tag, deadline)
> > +        begin
> > +          an_rpc = @server.request_call(@cq, request_call_tag, deadline)
> > +        rescue Core::CallError, RuntimeError => e
> > +          # can happen during server shutdown
>
> odd - this block is there because that's what I've observed. Are you sure
> that success=true and call=null occur for shutdown even when the server is
> shutdown from a separate thread ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/grpc/grpc/pull/1735/files#r30939122>.
>